agent/README.md

```diff
@@ -36,164 +35,178 @@ The following options are available:
       the current JVM session.
     - `[GET] /revision` Returns the current revision used for uploading to Teamscale.
-    - `[PUT] /revision` Sets the revision to use for uploading to Teamscale. The revision must be in the request body in plain text.
+    - `[PUT] /revision` Sets the revision to use for uploading to Teamscale. The revision must be in the request body in
+      plain text.
     - `[GET] /commit` Returns the current commit used for uploading to Teamscale.
-    - `[PUT] /commit` Sets the commit to use for uploading to Teamscale. The commit must be in the request body in plain thext in the format: branch:timestmap
-- `artifactory-url`: the HTTP(S) url of the artifactory server to upload the reports to.
-   The URL may include a subpath on the artifactory server, e.g. `https://artifactory.acme.com/my-repo/my/subpath`.
+    - `[PUT] /commit` Sets the commit to use for uploading to Teamscale. The commit must be in the request body in plain
+      thext in the format: branch:timestmap
+- `artifactory-url`: the HTTP(S) url of the artifactory server to upload the reports to. The URL may include a subpath
+  on the artifactory server, e.g. `https://artifactory.acme.com/my-repo/my/subpath`.
+- `artifactory-partition`: the partition name that is parsed later on via Teamscale Artifactory connector options.
 - `artifactory-user` (required for artifactory): The name of an artifactory user with write access.
 - `artifactory-password` (required for artifactory): The password of the user.
 - `artifactory-api-key` (alternative to `artifactory-user` and `artifactory-password`) The API key for artifactory from
-  a user with write access (c.f. [Artifactory Documentation](https://www.jfrog.com/confluence/display/JFROG/User+Profile#UserProfile-APIKey))
-- `artifactory-zip-path` (optional): The path within the stored ZIP file where the reports are stored.
-   Default is to store at root level.
-   This can be used to encode e.g. a partition name that is parsed later on via Teamscale Artifactory connector options. 
-- `artifactory-git-properties-jar` (optional): Specify a Jar to search a `git.properties` file within.
-   If not specified, Git commit information is extracted from the first found `git.properties` file.
-   See `teamscale-git-properties-jar` for details. 
+  a user with write access (
+  c.f. [Artifactory Documentation](https://www.jfrog.com/confluence/display/JFROG/User+Profile#UserProfile-APIKey))
+- `artifactory-zip-path` (optional): The path within the stored ZIP file where the reports are stored. Default is to
+  store at root level. This can be used to encode e.g. a partition name that is parsed later on via Teamscale
+  Artifactory connector options.
+- `artifactory-path-suffix` (optional): The path within the storage location between
+  the `<root-path>/uploads/<branch name>/<commit timestamp>[-<commit hash>]?/<upload partition>/<artifact type>` and the
+  uploaded artifact.
```

Comment:
It is unclear to me what kind of value I should pass in as a user here. Is it `uploads`? Is it the whole pattern that is shown here? Is it something that would come even after that presented pattern?

On a related topic: We should document somewhere (CHANGELOG) maybe how the upgrade path for existing users looks like. Do we recommend to adjust the configuration in Teamscale and use the new default format? How would the user need to configure the agent to keep the old behavior (in the valid case that the customer does not want to loose all old uploads)...

Reply:
Regarding the second comment: Currently it's not possible to edit the default. There's a feature ticket in the backlog but it's not implemented yet.

Reply:
@karottenreibe What is the strategy from your point of view that we should push towards? I would just want to avoid that this introduces such a breaking change that existing users cannot use any newer version of the agent without loosing all previous data.

Reply:
@andipunz We definitely need to state somewhere what this "default path" actually is and how to configure it in Teamscale manually even though it will be the default in the next Teamscale versions. Without this information it is impossible (without reading the code) to find out how to fit those pieces together.

Reply:
I implemented a legacy-upload option in order to keep the new behavior. I've just checked the issues regarding the support in TS itself. They are still in dev. Should we provide somewhere in the docs (here? TS-docs) how the extraction pattern needs to be modified to use this new version?

Reply:
Definitely yes. (See my other comment for more details)